sql: TestDistSQLFlowsVirtualTables failed

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11093936?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11093936?buildTab=artifacts#/) on master @ [51aa6257c063f60fb76044f9cccc47e259930d00](https://github.com/cockroachdb/cockroach/commits/51aa6257c063f60fb76044f9cccc47e259930d00):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 59003 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* created by github.com/cockroachdb/pebble/record.NewLogWriter
* 	github.com/cockroachdb/pebble/record/external/com_github_cockroachdb_pebble/record/log_writer.go:354 +0x3f0
* 
* goroutine 59018 [chan receive, 2 minutes]:
* github.com/cockroachdb/cockroach/pkg/sql_test.TestDistSQLFlowsVirtualTables.func4({0x403bb6c5a0?, 0x739b280?}, 0x40354fbec0?)
* 	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/crdb_internal_test.go:564 +0x184
* github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).SendWithWriteBytes(0x4019093900, {0x739b280?, 0x40354fbe90?}, 0x403bb6c5a0)
* 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_send.go:173 +0x3fc
* github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).SendWithWriteBytes(0x4018d34000, {0x739b280?, 0x40354fbe60?}, 0x403bb6c5a0)
* 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_send.go:193 +0x44c
* github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).SendWithWriteBytes(0x402dc9f9d0?, {0x739b280, 0x40354fbe60}, 0x403bb6c5a0)
* 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/stores.go:202 +0x9c
* github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal(0x4035ae1c00, {0x739b280?, 0x40354fbe30?}, {0x401a382360?}, 0x403bb6c5a0)
* 	github.com/cockroachdb/cockroach/pkg/server/node.go:1301 +0x2e4
* github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch(0x4035ae1c00, {0x739b280, 0x40354fbd70}, 0x403bb6c5a0)
* 	github.com/cockroachdb/cockroach/pkg/server/node.go:1432 +0x234
* github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler.func1({0x739b280, 0x40354fbd70}, {0x5af93a0?, 0x403bb6c5a0})
* 	github.com/cockroachdb/cockroach/pkg/kv/kvpb/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10277 +0x74
* github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.ServerInterceptor.func1({0x739b280, 0x40354fbd70}, {0x5af93a0, 0x403bb6c5a0}, 0x40062dc120, 0x401f9102a0)
* 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:97 +0x1b4
* google.golang.org/grpc.getChainUnaryHandler.func1({0x739b280, 0x40354fbd70}, {0x5af93a0, 0x403bb6c5a0})
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163 +0xa0
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3({0x739b280, 0x40354fbd70}, {0x5af93a0, 0x403bb6c5a0}, 0x40062dc120?, 0x4005cfb1c0)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:167 +0x88
* google.golang.org/grpc.getChainUnaryHandler.func1({0x739b280, 0x40354fbd70}, {0x5af93a0, 0x403bb6c5a0})
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163 +0xa0
* github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({0x4005c84000?, {{0x5780080?}, {0x73d9a20?, 0x4019016030?}}}, {0x739b280, 0x40354fbd70}, {0x5af93a0, 0x403bb6c5a0}, 0x40062dc120, 0x4005cfb140)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:105 +0x1ec
* google.golang.org/grpc.getChainUnaryHandler.func1({0x739b280, 0x40354fbd70}, {0x5af93a0, 0x403bb6c5a0})
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1163 +0xa0
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1({0x739b280?, 0x40354fbd70?})
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:134 +0x3c
* github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0x40039fd7c0, {0x739b280, 0x40354fbd70}, {0xc04ddc?, 0x38?}, 0x400b5b5910)
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:336 +0xac
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1({0x739b280?, 0x40354fbd70?}, {0x5af93a0?, 0x403bb6c5a0?}, 0x40062dc120?, 0x401f9102a0?)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:132 +0x7c
* google.golang.org/grpc.chainUnaryInterceptors.func1({0x739b280, 0x40354fbd70}, {0x5af93a0, 0x403bb6c5a0}, 0x400a1979f8?, 0xc05988?)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1154 +0x88
* github.com/cockroachdb/cockroach/pkg/kv/kvpb._Internal_Batch_Handler({0x5acbda0?, 0x4035ae1c00}, {0x739b280, 0x40354fbd70}, 0x400cf18070, 0x40334532a0)
* 	github.com/cockroachdb/cockroach/pkg/kv/kvpb/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvpb/kvpb_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvpb/api.pb.go:10279 +0x134
* google.golang.org/grpc.(*Server).processUnaryRPC(0x4001f3ad20, {0x73ed9e0, 0x40367ac820}, 0x403bb6c480, 0x401960df50, 0xa820800, 0x0)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1336 +0xb68
* google.golang.org/grpc.(*Server).handleStream(0x4001f3ad20, {0x73ed9e0, 0x40367ac820}, 0x403bb6c480, 0x0)
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1704 +0x840
* google.golang.org/grpc.(*Server).serveStreams.func1.2()
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:965 +0x84
* created by google.golang.org/grpc.(*Server).serveStreams.func1
* 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:963 +0x294
* 
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30186

--- a/pkg/sql/crdb_internal_test.go
+++ b/pkg/sql/crdb_internal_test.go
@@ -636,15 +636,11 @@ func TestDistSQLFlowsVirtualTables(t *testing.T) {
 	const clusterScope = "cluster"
 	const nodeScope = "node"
 	getNum := func(db *sqlutils.SQLRunner, scope string) int {
-		querySuffix := fmt.Sprintf("FROM crdb_internal.%s_distsql_flows", scope)
-		// Check that all remote flows (if any) correspond to the expected
-		// statement.
-		stmts := db.QueryStr(t, "SELECT stmt "+querySuffix)
-		for _, stmt := range stmts {
-			require.Equal(t, query, stmt[0])
-		}
+		// Count the number of flows matching our target query. Note that there
+		// could be other flows in the table for the internal operations.
+		countQuery := fmt.Sprintf("SELECT count(*) FROM crdb_internal.%s_distsql_flows WHERE stmt = '%s'", scope, query)
 		var num int
-		db.QueryRow(t, "SELECT count(*) "+querySuffix).Scan(&num)
+		db.QueryRow(t, countQuery).Scan(&num)
 		return num
 	}
 	for nodeID := 0; nodeID < numNodes; nodeID++ {

